### PR TITLE
Add code for face-recognition tutorial

### PR DESCRIPTION
**Pull request by martin-martin**
Relates to PR #691 on /tutorial-drafts
Moving the code here to avoid linter issues in the text repo.

**Where to put new files:**

- New files should go into a top-level subfolder, named after the article slug. For example: `my-awesome-article`

**How to merge your changes:** 

1. [Make sure the CI code style tests all pass (+ run the automatic code formatter if necessary).](https://github.com/realpython/materials/blob/master/README.md)
2. Find an RP Team member on Slack and ask them to review & approve your PR.
3. Once the PR has one positive ("approved") review, GitHub lets you merge the PR.
4. 🎉

### CODE REVIEW

File: face-recognition/source_code/detector.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            fill=(255, 255, 255, 255),
+        )
+
+    del draw
+    pillow_image.show()
+
+
+def validate(model: str = "hog"):
+    for filepath in pathlib.Path("validation").glob("**/*"):
+        recognize_faces(image_location=str(filepath), model=model)
```
Comment (kylestratis) [inline face-recognition/source_code/detector.py:125]:
```suggestion
        if filepath.is_file():
            recognize_faces(image_location=str(filepath), model=model)
```

**Review by kylestratis — CHANGES_REQUESTED** [review]
Made a suggested change to bring the code in line with the article code.

**Comment by martin-martin** [comment]
Adding a link here to the relevant conversation on the merged PR to make it easier to find:

https://github.com/realpython/materials/pull/369#issuecomment-1454056471

**Comment by kylestratis** [comment]
Made all suggested changes, all that was left were lines that the linter complains about, but black does not.

**Review by gahjelle — APPROVED** [review]
Great job @kylestratis and @martin-martin :+1: 

I did a quick update to use pathlib paths also for the encoding locations and a small update in step 6 where I think there was a bit of old code lying around :)

I also deleted the `source_code_step_7` folder since it's identical to `source_code_final`. Technically, we could even delete the step 6 folder, but I think it's okay to keep it around to avoid any confusion.

**Pull request merged**